From: Simon Richter <Simon.Richter@hogyros.de>                                                                                                                       
Subject: Re: error: implicit declaration of function 'cpu_die'                                                                                                       

arch/ppc/kernel/idle.c includes linux/smp.h, which includes asm-smp.h
only if CONFIG_SMP is defined.
As a result, cpu_die remains undefined for non-SMP builds.


  CC      arch/ppc/kernel/idle.o
arch/ppc/kernel/idle.c: In function 'default_idle':
arch/ppc/kernel/idle.c:58: error: implicit declaration of function 'cpu_die'
make[1]: *** [arch/ppc/kernel/idle.o] Error 1


Signed-off-by: Olaf Hering <olh@suse.de>

Index: linux-2.6.14/include/asm-ppc/smp.h
===================================================================
--- linux-2.6.14.orig/include/asm-ppc/smp.h	2005-12-01 13:16:51.000000000 +0100
+++ linux-2.6.14/include/asm-ppc/smp.h	2005-12-01 13:17:44.473289078 +0100
@@ -60,7 +60,6 @@ extern int smp_hw_index[];
 
 #else /* !(CONFIG_SMP) */
 
-static inline void cpu_die(void) { }
 #define get_hard_smp_processor_id(cpu) 0
 #define set_hard_smp_processor_id(cpu, phys)
 #define hard_smp_processor_id() 0
Index: linux-2.6.14/include/linux/smp.h
===================================================================
--- linux-2.6.14.orig/include/linux/smp.h	2005-12-01 13:16:51.000000000 +0100
+++ linux-2.6.14/include/linux/smp.h	2005-12-01 13:17:07.197335638 +0100
@@ -99,6 +99,7 @@ void smp_prepare_boot_cpu(void);
 static inline void smp_send_reschedule(int cpu) { }
 #define num_booting_cpus()			1
 #define smp_prepare_boot_cpu()			do {} while (0)
+static inline void cpu_die(void) { }
 
 #endif /* !SMP */
 
